atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchManagerImpl.java

```diff
@@ -70,6 +70,7 @@ public void close() {
 
     @Override
     public void registerWatches(Set<LockWatchReferences.LockWatchReference> newLockWatches) {
+        lockWatchReferences.clear();
```

Comment:
the way this works in combination with the javadoc is strange (it seems like register should definitely be commutative and distributive over union based on its javadoc; that it's not is a bit weird and so needs calling out I think, at least internally). I'd suggest we either
- keep this endpoint the way it is, and have a new endpoint called `registerPreciselyWatches` or something like that that makes it apparent that the client will no longer request previously registered things to be watched.
- or just rename this to `registerPreciselyWatches`, if the previous behaviour is not used anywhere